src/test/util/net.h

```diff
@@ -54,20 +54,19 @@ struct ConnmanTestMsg : public CConnman {
     void AddTestNode(CNode& node)
     {
         LOCK(m_nodes_mutex);
-        m_nodes.push_back(&node);
+        m_nodes.push_back(std::shared_ptr<CNode>(&node));
```

Comment:
This looks suspicious. No custom deleter is passed to the `shared_ptr` constructor, so it defaults to [`default_delete`](https://en.cppreference.com/w/cpp/memory/default_delete) which will `delete ptr;` eventually. Is the argument `CNode& node` guaranteed to be heap allocated? Otherwise:

```cpp
m_nodes.push_back(std::shared_ptr<CNode>(&node, [](CNode*){}));
```

Reply:
Yes, the argument `CNode& node` is always heap allocated. Not sure about "guaranteed". If not heap allocated, then this will be a gross error, resulting in an immediate crash or an error report by Valgrind or a memory sanitizer.

Note that the semantic of this piece of code is not changed by this PR. It was the same before as well - it would eventually `delete` the object.